test/e2e/e2e.go

```diff
@@ -46,86 +44,18 @@ import (
 
 	// ensure auth plugins are loaded
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	// ensure that cloud providers are loaded
+	_ "k8s.io/kubernetes/test/e2e/framework/providers/aws"
```

Comment:
Do we really need to hard smash in provider code?  
What are we using?  and given that they are moving out of tree what do we need other then a string? 

/cc @dims

Reply:
looks like we are only referring to aws here because some of the CI jobs run the AWS specific tests, so we'll probably fail on say on vmware specific tests (but won't know until someone tries to run them)

Reply:
test/e2e has traditionally contained tests that only work on certain providers. Is that really changing?

In this PR, I am only trying to untangle the code so that the the provider-specific tests are separate, optional packages. But I don't want to break the existing testing, so yes, the test/e2e.test binary still needs to import the same support code for different providers as before. For example, `_ "k8s.io/kubernetes/test/e2e/framework/providers/aws"` makes is possible to say `--provider=aws`.

Reply:
So I'm pretty against the idea of bin smashing in provider specifics into the main framework. In the long run, this is antithetical to where we want to go with provider code.  

If anything we want to excise it and wrap any kube specific are with an extension label, such as [storage]